panther_analysis_tool/rule.py

```diff
@@ -659,11 +679,34 @@ def _run_command(self, function: Callable, event: PantherEvent, expected_type: A
                 return result
             if not isinstance(result, list) or not all(isinstance(x, (str, bool)) for x in result):
                 raise FunctionReturnTypeError(
-                    "rule [{}] function [{}] returned [{}], expected a list".format(
-                        self.rule_id, function.__name__, type(result).__name__
+                    "detection [{}] function [{}] returned [{}], expected a list".format(
+                        self.detection_id, function.__name__, type(result).__name__
                     )
                 )
         return result
 
     def _is_function_defined(self, name: str) -> bool:
         return hasattr(self._module, name)
+
+
+class Rule(Detection):
+    """Panther rule metadata and imported module."""
+
+    # default detection types for rules
+    default_detection_type = "rule"
```

Comment:
Aha! I didn't know that this would satisfy the abstract property requirement, but it seems it does. The only drawback versus implementing them as property getters, is that someone can actually modify them more easily, as they seem as public class attributes. Not a really major issue though I believe, this will actually be just a bit more performant.